GenerateMacroDefinitionsFile.cmake

```diff
@@ -30,6 +30,10 @@ FILE(APPEND ${GENERATED_G3_DEFINITIONS} "// CMake induced definitions below. See
 
 FOREACH(definition ${G3_DEFINITIONS} )
    FILE(APPEND ${GENERATED_G3_DEFINITIONS} "#define ${definition}\n")
+   if(definition MATCHES ".* .*")
```

Comment:
@AndreasSchoenle : agreed. The path forward is to use the generated header file.

Since I cannot reproduce the issue (but have my suspicions) I'll let @jmelion explain how to reproduce this before closing the pull request. I need to better understand the circumstances before taking action.

1. Is CMake not used?
2. Compiler, platform, build steps?
3. Is it G3log that fails to compile or is it code linking with G3log?